src/pool/srv_pool.c

```diff
@@ -4266,6 +4414,180 @@ ds_pool_update(uuid_t pool_uuid, crt_opcode_t opc,
 	return rc;
 }
 
+static int
+ds_pool_extend_internal(struct rdb_tx *tx, struct pool_svc *svc,
+		uint32_t nnodes, uuid_t target_uuids[],
+		d_rank_list_t *rank_list, uint32_t ndomains,
+		int32_t *domains, bool *updated_p, uint32_t *map_version_p,
+		struct rsvc_hint *hint)
+{
+	struct pool_buf		*map_buf;
+	struct pool_map		*map = NULL;
+	uint32_t		map_version;
+	uuid_t			*uuids = NULL;
+	bool			updated = false;
+	int			ntargets;
+	int			rc;
+
+	ntargets = nnodes * dss_tgt_nr;
+
+	/* Create a temporary pool map based on the last committed version. */
+	rc = read_map(tx, &svc->ps_root, &map);
+	if (rc != 0)
+		return rc;
+
+	map_version = pool_map_get_version(map) + 1;
```

Comment:
Hmm, all new added servers/targets will share the same version. which does not match the fail/exclude process, where the version will be updated for each individual targets.  Is it ok for adding process?

Reply:
I don't think it will be a problem because they all start out at 0 as well when creating the pool map initially. 
During failure they should still be updated the same way they normally are.